### PR TITLE
Automatic testing on MacOS and Windows

### PR DESCRIPTION
**Pull request by sarathms**
Fixes ooni/probe#1350
Fixes ooni/probe#1356

### CODE REVIEW

**Comment by sarathms** [comment]
Rebasing with latest master.

File: main/index.js [hunk 1]
```diff
@@ -31,14 +37,11 @@ require('electron-debug')({
...
 autoUpdater.logger = log
 autoUpdater.logger.transports.file.level = 'info'
-log.info('App starting...')
+
 
 // To prevent garbage collection of the windows
 let windows = null
 
-// Set the application name
-app.name = 'OONI Probe'
```
Comment (hellais) [inline main/index.js:40]: Why was this removed?

File: main/utils/autorun/schedule.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
+const { join } = require('path')
+const { initConfigFile } = require('../config')
+const { getAutorunHomeDir } = require('../paths')
+
+// Initialize OONI_HOME for autorun by generating a fresh config file
+// in the OONI_HOME for autorun
+// Note: initConfigFile in turn initializes Sentry (again) with no extra effect
+initConfigFile({ configFilePath: join(getAutorunHomeDir(), 'config.json') })
```
Comment (hellais) [inline main/utils/autorun/schedule.js:8]: How is the config file for the autorun kept in sync with the config file for the normal OONI Probe?

File: main/utils/autorun/mac-scheduler.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        OONI_HOME_autorun: getAutorunHomeDir()
+      })
+      const taskPlistPath = join(
+        app.getPath('home'), 'Library', 'LaunchAgents',
+        `${taskname}.plist`
+      )
+      writeFile(taskPlistPath, taskPlistStr).then(() => {
+        log.debug(`Autorun task plist file created: ${taskPlistPath}`)
+        log.debug(`Enable task with: launchctl load ${taskPlistPath}`)
+        const cmd = launchctl('load', [taskPlistPath])
```
Comment (hellais) [inline main/utils/autorun/mac-scheduler.js:36]: Are you sure this is the right command? In probe-cli we are instead doing:
```
launchctl enable && launchctl bootstrap
```

File: main/utils/autorun/mac-scheduler.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        log.debug(`Enable command result: ${cmd.stdout.toString().trim()}`)
+        resolve()
+      }).catch(e => {
+        reject(`Failed to create plist file: ${e.message}`)
+      })
+    })
+  },
+  delete: function (taskname) {
+    return new Promise((resolve, reject) => {
+      // launchctl('unload', [taskPlistPath])
```
Comment (hellais) [inline main/utils/autorun/mac-scheduler.js:51]: Here it should delete the file and call `bootout`

File: main/utils/autorun/taskTemplateWin.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+const getStartBoundary = () => {
+  const now = new Date()
+  // In one minute
+  now.setMinutes(now.getMinutes() + 1)
+
+  // use Date.toISOString() and account for timezone offset
+  // StartBoundary format - YYYY-MM-DDTHH:mm:ss
+  // Date.toISOString()   - YYYY-MM-DDTHH:mm:ss.sssZ
+  now.setMinutes(now.getMinutes() - now.getTimezoneOffset())
```
Comment (hellais) [inline main/utils/autorun/taskTemplateWin.js:11]: Why are you calling `setMinutes` twice?

File: main/index.js (hunk 1 above)
Comment (sarathms) [reply main/index.js:40]: We will use the value `productName` in `package.json` to keep it consistent. It already uses the same value.

File: main/utils/autorun/schedule.js (hunk 2 above)
Comment (sarathms) [reply main/utils/autorun/schedule.js:8]: I lost sight of all the usecases for which we wanted to `sync` and not `copy`. I have to do this now.

File: main/utils/autorun/mac-scheduler.js (hunk 3 above)
Comment (sarathms) [reply main/utils/autorun/mac-scheduler.js:56]: In most places I read, it is recommended to use `load` because it registers the task and starts it right away.
https://medium.com/swlh/how-to-use-launchd-to-run-services-in-macos-b972ed1e352
https://www.maketecheasier.com/use-launchd-run-scripts-on-schedule-macos/
https://apple.stackexchange.com/questions/29056/launchctl-difference-between-load-and-start-unload-and-stop

File: main/utils/autorun/mac-scheduler.js (hunk 4 above)
Comment (sarathms) [reply main/utils/autorun/mac-scheduler.js]: This is now done.

File: main/utils/autorun/mac-scheduler.js (hunk 3 above)
Comment (hellais) [reply main/utils/autorun/mac-scheduler.js:56]: @bassosimone maybe we want to use the same behaviour on probe-cli then.

Comment (bassosimone) [reply main/utils/autorun/mac-scheduler.js:56]: I think we should not. The `load` command is deprecated and may not work. I think it's safer that `probe-desktop` does what `probe-cli` does, which is what `brew services` does.

Comment (sarathms) [reply main/utils/autorun/mac-scheduler.js:56]: Thanks @bassosimone for showing how `load/unload` is not the recommended way to do this. I will go ahead and change this.

**Review by hellais — APPROVED** [review]
LGTM
